substrate/frame/examples/authorization-tx-extension/src/lib.rs

```diff
@@ -0,0 +1,155 @@
+//! integration in a `TransactionExtension` pipeline.
+//!
+//! The [TransactionExtension](sp_runtime::traits::TransactionExtension) used in this example is
+//! [AuthorizeCoownership](extensions::AuthorizeCoownership). If activated, the extension will
+//! authorize 2 signers as coowners, with a [coowner origin](pallet_coownership::Origin) specific to
+//! the [coownership example pallet](pallet_coownership), by validating a signature of the rest of
+//! the transaction from each party. This means any extensions after ours in the pipeline, their
+//! implicits and the actual call. The extension pipeline used in our example checks the genesis
+//! hash, transaction version and mortality of the transaction after the `AuthorizeCoownership` runs
+//! as we want these transactions to run regardless of what origin passes through them and/or we
+//! want their implicit data in any signature authorization happening earlier in the pipeline.
+//!
+//! In this example, aside from the [AuthorizeCoownership](extensions::AuthorizeCoownership)
+//! extension, we use the following pallets:
+//! - [pallet_coownership] - provides a coowner origin and the functionality to authorize it.
+//! - [pallet_assets] - a dummy asset pallet that tracks assets, identified by an
+//!   [AssetId](pallet_assets::AssetId), and their respective owners, which can be either an
+//!   [account](pallet_assets::Owner::Single) or a [pair of owners](pallet_assets::Owner::Double).
+//!
+//! Assets are created in [pallet_assets] using the
+//! [create_asset](pallet_assets::Call::create_asset) call, which accepts traditionally signed
+//! origins (a single account) or coowner origins, authorized through the
+//! [CoownerOrigin](pallet_assets::Config::CoownerOrigin) type.
+//!
+//! ### Example
+#![doc = docify::embed!("src/tests.rs", create_coowned_asset_works)]
+//!
+//! This example does not focus on any pallet logic or syntax, but rather on `TransactionExtension`
+//! functionality. The pallets used are just skeletons to provide storage state and custom origin
+//! choices and requirements, as shown in the examples. Any weight and/or
+//! transaction fee is out of scope for this example.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub mod extensions;
+
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+extern crate alloc;
+
+use frame_support::pallet_prelude::*;
+use frame_system::pallet_prelude::*;
+
+#[frame_support::pallet(dev_mode)]
+pub mod pallet_coownership {
+	use super::*;
+	use frame_support::traits::OriginTrait;
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// The aggregated origin which the dispatch will take.
+		type RuntimeOrigin: OriginTrait<PalletsOrigin = Self::PalletsOrigin>
+			+ From<Self::PalletsOrigin>
+			+ IsType<<Self as frame_system::Config>::RuntimeOrigin>;
+
+		/// The caller origin, overarching type of all pallets origins.
+		type PalletsOrigin: From<Origin<Self>> + TryInto<Origin<Self>, Error = Self::PalletsOrigin>;
+	}
```

Comment:
just for information this would be the implementation without associated types:
```diff
diff --git a/substrate/frame/examples/authorization-tx-extension/src/lib.rs b/substrate/frame/examples/authorization-tx-extension/src/lib.rs
index 60d198bdebf..accb97f3b72 100644
--- a/substrate/frame/examples/authorization-tx-extension/src/lib.rs
+++ b/substrate/frame/examples/authorization-tx-extension/src/lib.rs
@@ -70,16 +70,20 @@ use frame_system::pallet_prelude::*;
 pub mod pallet_coownership {
        use super::*;
        use frame_support::traits::OriginTrait;
+       pub type PalletsOriginFor<T> = <<T as frame_system::Config>::RuntimeOrigin as OriginTrait>::PalletsOrigin;
 
-       #[pallet::config]
-       pub trait Config: frame_system::Config {
-               /// The aggregated origin which the dispatch will take.
-               type RuntimeOrigin: OriginTrait<PalletsOrigin = Self::PalletsOrigin>
-                       + From<Self::PalletsOrigin>
-                       + IsType<<Self as frame_system::Config>::RuntimeOrigin>;
 
-               /// The caller origin, overarching type of all pallets origins.
-               type PalletsOrigin: From<Origin<Self>> + TryInto<Origin<Self>, Error = Self::PalletsOrigin>;
+       #[pallet::config]
+       pub trait Config: frame_system::Config<
+               RuntimeOrigin:
+                       OriginTrait<
+                               PalletsOrigin:
+                                       From<Origin<Self>>
+                                       + TryInto<Origin<Self>, Error = PalletsOriginFor<Self>>
+                       >
+                       + From<PalletsOriginFor<Self>>
+       >
+       {
        }
 
        #[pallet::pallet]
```

Reply:
anyway this requires rust 1.79. and we are using 1.77